src/sentry/middleware/integrations/parsers/slack.py

```diff
@@ -82,19 +92,101 @@ class SlackRequestParser(BaseRequestParser):
     See: `src/sentry/integrations/slack/views`
     """
 
+    def build_loading_modal(self, external_id: str, title: str):
+        return {
+            "type": "modal",
+            "external_id": external_id,
+            "title": {"type": "plain_text", "text": title},
+            "blocks": [
+                {
+                    "type": "section",
+                    "block_id": "loading_block",
+                    "text": {"type": "mrkdwn", "text": "Loading..."},
+                }
+            ],
+        }
+
+    def handle_dialog(self, request, action, title):
+        decoded_body = parse_qs(request.body.decode(encoding="utf-8"))
+        if (payload_dict := decoded_body.get("payload")) is None:
+            raise ValueError("request format is incorrect")
+
+        payload = json.loads(payload_dict[0])
+
+        # we need to grab the action_ts to use as the external_id for the loading modal
+        # https://api.slack.com/reference/interaction-payloads/block-actions
+        action_ts = payload.get("actions")[0].get("action_ts")
```

Comment:
does this HAVE to be the `action_ts`, or are we just using it because it's basically guaranteed to be unique?